### ISSUE TITLE
USB installer images won't work with BIOS

### ISSUE THREAD

**Issue by LinuxKernel1** [opnsense/core#3528]
Used dd and rufus to create a bootable usb device (OPNsense-19.1.4-OpenSSL-vga-amd64.img), but refuses to boot on various machines with BIOS. After the boot devices is selected a blank screen appears for some seconds and you are back at the boot device selection.
On a PC with UEFI it works without a problem. 
Also tried version 18.7 & 18.1 with the same result.

The pfsense image(pfSense-CE-memstick-2.4.4-RELEASE-p3-amd64.img) works perfectly.

**Comment by LinuxKernel1** [comment]
Current situation after some more testing:
Machine 1 - BIOS: NOT WORKING
Machine 2 - BIOS: NOT WORKING
Machine 3 - BIOS: WORKING
Machine 4 - UEFI: WORKING
OPNsense-19.1.4-OpenSSL-vga-amd64.img and rufus were used.

Workaround:
1. Create a bootable media with rufus
2. Convert the GPT disk to MBR
3. Mark the EFISYS partition as "active"
--> WORKING on all BIOS PC's<--

Booting from GPT based images is intended to work on BIOS PC's as well, so there is still somewhere a problem.

**Comment by fawzy46** [comment]
> Current situation after some more testing:
> Machine 1 - BIOS: NOT WORKING
> Machine 2 - BIOS: NOT WORKING
> Machine 3 - BIOS: WORKING
> Machine 4 - UEFI: WORKING
> OPNsense-19.1.4-OpenSSL-vga-amd64.img and rufus were used.
> 
> Workaround:
> 
> 1. Create a bootable media with rufus
> 2. Convert the GPT disk to MBR
> 3. Mark the EFISYS partition as "active"
>    --> WORKING on all BIOS PC's<--
> 
> Booting from GPT based images is intended to work on BIOS PC's as well, so there is still somewhere a problem.

can you share more detailed steps, please 

converting to gpt format the hard drive right? so how we do it after creating a bootable media?

**Comment by LinuxKernel1** [comment]
> converting to gpt format the hard drive right?
After you wrote the image to your USB device it automatically will use GPT and to make it work you have to convert it to MBR. You just have to modify your USB device, your PC will be untouched.

> so how we do it after creating a bootable media?
You still can modify your bootable USB device after you have written the image onto it. There are various tools available for it.

**Comment by fawzy46** [comment]
> > converting to gpt format the hard drive right?
> 
> After you wrote the image to your USB device it automatically will use GPT and to make it work you have to convert it to MBR. You just have to modify your USB device, your PC will be untouched.
> 
> > so how we do it after creating a bootable media?
> 
> You still can modify your bootable USB device after you have written the image onto it. There are various tools available for it

I thought i have done something wrong as my usb always turn to gpt after burning. One last thing after  burning the img I can't open the usb from windows or display the files it contains is that normal?

**Comment by LinuxKernel1** [comment]
> I thought i have done something wrong as my usb always turn to gpt after burning. 
Usually depends on a few things, but in case of opnsense it's normal (predefined by the image).

> One last thing after burning the img I can't open the usb from windows or display the files it contains is that normal?
As far as I can remember I think it's normal.

**Comment by fawzy46** [comment]
Unfortunately I did what you said exactly but no success. There is no EFIsys partition on my usb drive

**Comment by AdSchellevis** [comment]
This issue has been automatically timed-out (after 180 days of inactivity).

For more information about the policies for this repository,
please read https://github.com/opnsense/core/blob/master/CONTRIBUTING.md for further details.

If someone wants to step up and work on this issue,
just let us know, so we can reopen the issue and assign an owner to it.

**Comment by erdincay** [comment]
I have the same issue. The problem is with the IMG. I have used DD and Popsicle. I am pretty sure that DD has no issues, and Popsicle hasn't yet let me down, writing an simple image. The ISO file (for DVD/CD etc.) is written perfectly, but it is not bootable.

**Comment by alexis-via** [comment]
Having the same pb: image with OPNsense-22.7-OpenSSL-vga-amd64.img doesn't work on a BIOS PC. OPNsense Nano image and pfsense 2.6 image boot fine.
I copied OPNsense-22.7-OpenSSL-vga-amd64.img to my USB key using dd. Then I used gdisk to convert GPT to MBR. Then I used fdisk to add the boot flag to the EFI partition (the first partition). After that, when I try to boot the USB key, BIOS shows a message "Invalid partition table".
